### PR TITLE
Add option to set the Certificate Authority (CA) Info option.

### PR DESCRIPTION
**Pull request by tgoetze**
This is used when verifying a peer. See CURLOPT_CAINFO.

### CODE REVIEW

**Comment by tgoetze** [comment]
I am not sure what happens if someone does not have openssl/ssl.h available--it could fail.
If someone sees this patch fail, I would be happy to use a AC_CHECK_HEADERS and #ifdef to either make the call a no-op, or to completely hide the new method in the Connection class.

File: source/connection.cc [hunk 1]
```diff
@@ -118,6 +118,19 @@ RestClient::Connection::SetUserAgent(const std::string& userAgent) {
...
+ * default restclient-cpp/RESTCLIENT_VERSION string
+ *
+ * @param caInfo - The path to a file holding the certificates used to
+ * verify the peer with. See CURLOPT_CAINFO
+ *
+ */
+void
+RestClient::Connection::SetCAInfo(const std::string& caInfo) {
+  this->caInfo = caInfo;
+}
```
Comment (mrtazz) [inline source/connection.cc]: I think something like `SetCAInfoFilePath` for the method and the corresponding instance variable make it more clear what the argument should actually be (filepath vs actual contents as string)

File: source/connection.cc [hunk 2]
```diff
@@ -118,6 +118,19 @@ RestClient::Connection::SetUserAgent(const std::string& userAgent) {
 }
 
 /**
+ * @brief set custom Certificate Authority (CA) path
+ * default restclient-cpp/RESTCLIENT_VERSION string
```
Comment (mrtazz) [inline source/connection.cc]: is this a copy and paste leftover?

**Comment by mrtazz** [comment]
This is awesome! Thank you for taking the time to submit this patch! I added some minor comments but it looks great overall.

File: source/connection.cc (hunk 1 above)
Comment (tgoetze) [inline source/connection.cc]: I considered it, but decided to keep consistent with libcurl naming (even though it is not as clear as I would have liked). If you prefer clarity of names in restclient-cpp even though they would not be consistent with libcurl, I would be happy to change--just commenting to get it considered.

Comment (mrtazz) [inline source/connection.cc]: Yup, thanks for thinking of this! I thought about it as well and I'm ok with diverging slightly from the libcurl naming to get better clarity into restclient-cpp.

Comment (tgoetze) [inline source/connection.cc]: Changed as requested.

**Comment by mrtazz** [comment]
Awesome! Thanks so much for the contribution!

**Pull request merged**